gdal/port/cpl_conv.h

```diff
@@ -358,9 +363,34 @@ class CPL_DLL CPLConfigOptionSetter
     char *m_pszOldValue;
     bool m_bRestoreOldValue;
 };
+}
 
 #endif /* def __cplusplus */
 //! @endcond
 
+#if defined(__cplusplus) && !defined(CPL_SUPRESS_CPLUSPLUS)
+
+extern "C++"
+{
+namespace cpl
```

Comment:
Woohoo!  namespaces for the win!

Reply:
Why this has to be in the public `cpl_conv.h` instead of a private header included in .cpp files only?

Reply:
@mloskot cpl::down_cast is used by the new inline conveniency OGRGeometry::toPoint(), toLineString() etc methods of ogr_geometry.h (which is already an installed header), so it needs to be somewhere visible.